File: output/deepseek_r1/avatar/direct_translation/python/java/failed_translation/codeforces_268_A.java

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class codeforces_268_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        List<String[]> list = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            String[] parts = scanner.nextLine().split(" ");
            list.add(parts);
        }
        int count = 0;
        for (String[] j : list) {
            for (String[] k : list) {
                if (j == k) {
                    continue;
                }
                if (j[0].length() > 0 && k[k.length - 1].length() > 0) {
                    if (j[0].charAt(0) == k[k.length - 1].charAt(0)) {
                        count++;
                    }
                }
            }
        }
        System.out.println(count);
    }
}

File: dataset/avatar/Python/Code/codeforces_268_A.py
n = int ( input ( ) )
q = [ ]
count = 0
for i in range ( n ) :
    a = input ( ).split ( )
    q.append ( a )
for j in q :
    for k in q :
        if k == j :
            continue
        elif j [ 0 ] == k [ len ( k ) - 1 ] :
            count += 1
        else :
            continue
print ( count )

